tests/Parallel_FilenameTest.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright 2016 Alexander Grund
+ *
+ * This file is part of libSplash.
+ *
+ * libSplash is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * libSplash is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libSplash.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+ #include <cppunit/extensions/HelperMacros.h>
+
+#include "splash/splash.h"
+#include <mpi.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+using namespace splash;
+
+class FilenameTest : public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE(FilenameTest);
+
+    CPPUNIT_TEST(testBaseName);
+    CPPUNIT_TEST(testFullName);
+
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+
+    FilenameTest();
+    virtual ~FilenameTest();
+
+private:
+    /**
+     * Tests that file naming is correct when passing only base name
+     */
+    void testBaseName();
+
+    /**
+     * Tests that file naming is correct when passing full name
+     */
+    void testFullName();
+
+    void runTest(const char* filename, const char* fullFilename);
+
+    bool fileExists(const std::string& filename);
+
+    IParallelDataCollector *dataCollector;
+    ColTypeInt ctInt;
+    int mpiRank;
+    static int initCt;
+};
+
+int FilenameTest::initCt = 0;
+
+CPPUNIT_TEST_SUITE_REGISTRATION(FilenameTest);
+
+#define HDF5_FILE "h5/testParallelFilename"
+#define HDF5_BASE_FILE HDF5_FILE "_2.h5"
+#define HDF5_FULL_FILE HDF5_FILE ".h5"
+
+#define MPI_SIZE_X 1
+#define MPI_SIZE_Y 1
+
+FilenameTest::FilenameTest()
+{
+    if(!initCt)
+    {
+        MPI_Init(NULL, NULL);
+        srand(time(NULL));
+        // Remove old files
+        unlink(HDF5_BASE_FILE);
+        unlink(HDF5_FULL_FILE);
+    }
+    initCt++;
+
+    int mpiSize;
+    MPI_Comm_rank(MPI_COMM_WORLD, &mpiRank);
+    MPI_Comm_size(MPI_COMM_WORLD, &mpiSize);
+
+    CPPUNIT_ASSERT(mpiSize == (MPI_SIZE_X * MPI_SIZE_Y));
+
+    dataCollector = new ParallelDataCollector(MPI_COMM_WORLD, MPI_INFO_NULL,
```

Comment:
shouldn't we test a `SerialDataCollector` read/write from a full filename, too?

E.g., expect exception on write with SerialDataCollector for single-file mode and allow `read` in same mode.